metrics/code_eval/README.md

```diff
@@ -0,0 +1,128 @@
+
+`references`: a list with a test for each prediction. Each test should evaluate the correctness of a code candidate.
+
+`k`: number of code candidates to consider in the evaluation. The default value is `[1, 10, 100]`.
+
+`num_workers`: the number of workers used to evaluate the candidate programs (The default value is `4`).
+
+`timeout`: The maximum time taken to produce a prediction before it is considered a "timeout". The default value is `3.0` (i.e. 3 seconds).
+
+```python
+from datasets import load_metric
+code_eval = load_metric("code_eval")
+test_cases = ["assert add(2,3)==5"]
+candidates = [["def add(a,b): return a*b", "def add(a, b): return a+b"]]
+pass_at_k, results = code_eval.compute(references=test_cases, predictions=candidates, k=[1, 2])
+```
+
+N.B.
+Before running this metric, you will need to run the following commands:
+```python
+import os
+os.environ["HF_ALLOW_CODE_EVAL"] = "1"` 
+```
+
+## Output values
+
+The Code Eval metric outputs two things:
+
+`pass_at_k`: a dictionary with the pass rates for each k value defined in the arguments.
+
+`results`: a dictionary with granular results of each unit test.
+
+### Values from popular papers
+The [original CODEX paper](https://arxiv.org/pdf/2107.03374.pdf) reported that the CODEX-12B model had a pass@k score of 28.8% at `k=1`, 46.8% at `k=10` and 72.3% at `k=100`. However, since the CODEX model is not open source, it is hard to verify these numbers. 
+
+
+
+## Examples 
+
+Full match at `k=1`:
+
+```python
+from datasets import load_metric
+code_eval = load_metric("code_eval")
+test_cases = ["assert add(2,3)==5"]
+candidates = [["def add(a, b): return a+b"]]
+pass_at_k, results = code_eval.compute(references=test_cases, predictions=candidates, k=[1])
+print(pass_at_k)
+{'pass@1': 1.0}
+```
+
+No match for k = 1:
+
+```python
+from datasets import load_metric
+code_eval = load_metric("code_eval")
+test_cases = ["assert add(2,3)==5"]
+candidates = [["def add(a,b): return a*b"]]
+pass_at_k, results = code_eval.compute(references=test_cases, predictions=candidates, k=[1])
+print(pass_at_k)
+{'pass@1': 0.0}
+```
+
+Partial match at k=1, full match at k=2:
+
+```python
+from datasets import load_metric
+code_eval = load_metric("code_eval")
+test_cases = ["assert add(2,3)==5"]
+candidates = [["def add(a, b): return a+b", "def add(a,b): return a*b"]]
+pass_at_k, results = code_eval.compute(references=test_cases, predictions=candidates, k=[1, 2])
+print(pass_at_k)
+{'pass@1': 0.5, 'pass@2': 1.0}
+```
+
+## Limitations and bias
+
+As per the warning included in the metric code itself:
+"This program exists to run untrusted model-generated code. Users are strongly encouraged not to do so outside of a robust security sandbox. The execution call in execution.py is deliberately commented out to ensure users read this disclaimer before running code in a potentially unsafe manner. See the comment in execution.py for more information and instructions."
```

Comment:
this is the full message in the code:

```suggestion
> This program exists to execute untrusted model-generated code. Although it is highly unlikely that model-generated code will do something overtly malicious in response to this test suite, model-generated code may act destructively due to a lack of model capability or alignment. Users are strongly encouraged to sandbox this evaluation suite so that it does not perform destructive actions on their host or network. For more  information on how OpenAI sandboxes its code, see the accompanying paper. Once you have read this disclaimer and taken appropriate precautions, uncomment the following line and proceed at your own risk:
```